### PR TITLE
Cleanup config initialization, add local config options

### LINKED ISSUE #5313: Local config data is cleared when ~/.cache mounted on tempfs
## Overview
Since the big update (new gui) keepassXC  very very rarely remembers the last database no matter whether I shutdown the desktop cleanly or with a three or four finger salute (KDE only ?). Before the update it remembered fine. It is enabled in the options. Remember and open on startup

As I only ever use one database my workaround has been , just now, to change it's user .desktop file to explicitly open the database

## Steps to Reproduce

1. Tell keepassXC to remember last database and open on start up.
2. Restart the computer or X by any method
3. Start KeepassXC

## Expected Behavior
Ask me for the password for the last used database instead of having to navigate to it's location almost every time.

## Actual Behavior
Have to navigate to it's location almost every time.

## Context
It occasionally works but I fail to see a pattern. I've tried making sure KeepassXC is closed cleanly but that doesn't make a difference AFAICT. 

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.6.1
Revision: 9a35bba

Qt 5.15.0
Debugging mode is disabled.

Operating system: Gentoo/Linux
CPU architecture: x86_64
Kernel: linux 5.8.3-gentoo-x86_64

Enabled extensions:
- SSH Agent
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.6

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linux/
Desktop Env: KDE
Windowing System: X11

### PR DESCRIPTION
**Pull request by droidmonkey**
* Fix #5313, allow specifying local config path using environment variable and command line flag
* Add command line flag `--localconfig <path>` to specify a file path to use for the local configuration settings.
* Add environment variable support to set config files paths: `KPXC_CONFIG` and `KPXC_CONFIG_LOCAL` to override default locations.
* Reorder startup sequence to load specified config files earlier to allow for theme settings and other early options to be picked up.
* Removed old command line option `--pw`, no longer used.

* Attempt a fix of application not closing when last window is gone. Only set `QApplication::setQuitOnLastWindowClosed(true)` when tray icon is enabled instead of always.

Replacement for #5325 

[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested on Windows

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)
- ✅ Refactor (significant modification to existing code)

### CODE REVIEW

**Comment by kotoko** [comment]
When using variable `KPXC_CONFIG_LOCAL` should I set `KPXC_CONFIG_LOCAL="~/.local/share"` or `KPXC_CONFIG_LOCAL="~/.local/share/keepassxc"`?

**Comment by droidmonkey** [comment]
Set it to whatever you want, but this points to a file not a directory.

**Comment by kotoko** [comment]
Oh... So I guess it should be `KPXC_CONFIG_LOCAL="~/.local/share/keepassxc/keepassxc.ini"`. Last question: will it work if `keepassxc` folder does not exist?

**Comment by droidmonkey** [comment]
Give it a try, test it out please

**Comment by kotoko** [comment]
OK. I tried to build keepassxc following readme (with cmake arguments `-DWITH_XC_ALL=OFF -DCMAKE_BUILD_TYPE=Release`) and got weird errors. For example:

```
/home/*****/build/CMakeFiles/CMakeTmp/src.cxx:2:44: error: too many arguments to function ‘size_t malloc_usable_size(void*)’
    2 |     int main() { malloc_usable_size(NULL, 0); return 0; }
      |                                            ^
In file included from /home/*****/build/CMakeFiles/CMakeTmp/src.cxx:1:
/usr/include/malloc.h:136:15: note: declared here
  136 | extern size_t malloc_usable_size (void *__ptr) __THROW;
      |               ^~~~~~~~~~~~~~~~~~
```

Given that I'm not interested in building and debugging keepassxc manually I'll just wait until this get merged to master and packaged by my linux distribution maintainers.

**Review by phoerious — APPROVED** [review]

**Comment by phoerious** [comment]
> Last question: will it work if keepassxc folder does not exist?

QSettings creates directories if they don't exist.

**Pull request merged**
